css-fonts-3/font-style-matching-manual.html

```diff
@@ -0,0 +1,2712 @@
+    font-stretch: ultra-expanded;
+    font-style: italic;
+    font-weight: 100;
+
+}
+
+
+
+    #ultraexpanded_italic_twohundred {
+    font-family: CSSMatchingTest;
+    font-stretch: ultra-expanded;
+    font-style: italic;
+    font-weight: 200;
+
+}
+
+
+
+    #ultraexpanded_italic_threehundred {
+    font-family: CSSMatchingTest;
+    font-stretch: ultra-expanded;
+    font-style: italic;
+    font-weight: 300;
+
+}
+
+
+
+    #ultraexpanded_italic_fourhundred {
+    font-family: CSSMatchingTest;
+    font-stretch: ultra-expanded;
+    font-style: italic;
+    font-weight: 400;
+
+}
+
+
+
+    #ultraexpanded_italic_fivehundred {
+    font-family: CSSMatchingTest;
+    font-stretch: ultra-expanded;
+    font-style: italic;
+    font-weight: 500;
+
+}
+
+
+
+    #ultraexpanded_italic_sixhundred {
+    font-family: CSSMatchingTest;
+    font-stretch: ultra-expanded;
+    font-style: italic;
+    font-weight: 600;
+
+}
+
+
+
+    #ultraexpanded_italic_sevenhundred {
+    font-family: CSSMatchingTest;
+    font-stretch: ultra-expanded;
+    font-style: italic;
+    font-weight: 700;
+
+}
+
+
+
+    #ultraexpanded_italic_eighthundred {
+    font-family: CSSMatchingTest;
+    font-stretch: ultra-expanded;
+    font-style: italic;
+    font-weight: 800;
+
+}
+
+
+
+    #ultraexpanded_italic_ninehundred {
+    font-family: CSSMatchingTest;
+    font-stretch: ultra-expanded;
+    font-style: italic;
+    font-weight: 900;
+
+}
+
+
+
+
+</style>
+</head>
+<body>
+    <h1>CSS Font Style Matching Test</h1>
+    <p>This test tests for steps <a href="http://www.w3.org/TR/css3-fonts/#font-style-matching">4. a-c of the font style
+    matching algorithm</a>. Stretch, style and weight information specified on the left-hand side should match the
+    information on the right-hand side.</p>
+<p>This test works by declaring @font-face rules for each combination of stretch, style and weight and refering to a
+    different src url for each such rules. The font that the src url points to has a special property: it renders
+    ligatures for the words &quot;stretch&quot;, &quot;style&quot; and &quot;weight&quot; according to the values
+    specified in its filename, e.g. CSSMatchingTest_condensed_normal_fourhundred.ttf renders the word/ligature
+    &quot;condensed&quot; for the word stretch, &quot;normal&quot; instead of the word style, and &quot;400&quot; for
+    the word weight. It also generates style declarations using IDs using the same naming scheme of stretch, style and weight.</p>
+<p>The HTML then iterates over all the combinations and in each line refers to a new style. In the right column, the
+    selected font can be verified by checking whether the correct ligatures were rendered. If they match, the user
+    agent matched the right font-face rule and loaded the correct respective ttf file for rendering.</p>
+<table>
+<tr><td >normal_normal_onehundred</td><td id="normal_normal_onehundred">stretch style weight</td></tr>
```

Comment:
I'm a little new to this test infrastructure, but where are the expected results? If this is a reftest, don't you have the problem where there is so much content that it will overflow the viewport and not be tested? If this isn't a reftest, how are you using ligatures to test the results?